### ISSUE TITLE
Does BatchNorm make performance uncertain even worse?

### ISSUE THREAD

**Issue by lscelory** [carrierlxk/COSNet#10]
Hi,
According to your paper, the batch size in training is set to 8. I also checked your GPU **Nvidia TITAN Xp** which memory is 12GB one device. In my own training, I notice I only can set 2 batch(2 pairs, equal to 4 images) one gpu because the deeplabv3 backbone need much GPU memory. If I set batchsize=8, I have to use 4 gpus in data parallel way. However, for BatchNorm layer it is equivalent to batchsize=2(minibatch=2). Each gpu calculate its own 2 batch tensor's norm. And according to my experiment, I use batchsize=8(minibatch=2) and 4 gpus, my test result on DAVIS16-test is unpredictable.(from ~68% to ~73%) My best result is only ~73% and it is much worse than your paper reporting. Hoping you can give me more advices. Thankyou!!!

**Comment by carrierlxk** [comment]
Hi, you can fix the batch norm layer. In addition, to reduce the gpu memory, you can reduce the input image size from 473 to 378.

**Comment by lscelory** [comment]
> Hi, you can fix the batch norm layer. In addition, to reduce the gpu memory, you can reduce the input image size from 473 to 378.

Thank you for your advice! I will try these settings.

**Comment by NoaHimesaka** [comment]
> > Hi, you can fix the batch norm layer. In addition, to reduce the gpu memory, you can reduce the input image size from 473 to 378.
> 
> Thank you for your advice! I will try these settings.

I'm sorry to bother you
But I'm curious, do you get a better result?
I've met the same question with you, I can only set batch_size=4 and change the input size to 378 using 2 GPU 1080ti

**Comment by lscelory** [comment]
According to my exps, it seems batch size is not the determinant factor as long as you have a strong pre-trained model.

**Comment by NoaHimesaka** [comment]
> According to my exps, it seems batch size is not the determinant factor as long as you have a strong pre-trained model.

Thank you!
